app/jobs/comments/calculate_score_job.rb

```diff
@@ -6,7 +6,7 @@ def perform(comment_id)
       comment = Comment.find_by(id: comment_id)
 
       comment&.update_columns(score: BlackBox.comment_quality_score(comment), spaminess_rating: BlackBox.calculate_spaminess(comment))
-      comment.root.save unless comment.is_root?
+      comment.root.save unless comment&.is_root?
```

Comment:
I still think the guard clause would be more explicit.
On this line in case `comment` is `nil` `comment&.is_root?` will return `nil` and the `comment.root.save` will be executed which will lead to an error.